src/Italy/Mews.Fiscalizations.Italy/Uniwix/Communication/UniwixClient.cs

```diff
@@ -41,32 +43,25 @@ public async Task<SendInvoiceResult> SendInvoiceAsync(ElectronicInvoice invoice)
                 { new ByteArrayContent(invoiceFile.Data), "fattura", invoiceFile.FileName }
             };
 
-            try
-            {
-                var result = await PostAsync<PostInvoiceResponse>(url, content);
-                return new SendInvoiceResult(result.FileId, result.Message);
-            }
-            catch (UniwixException e) when (e.Code == (int)UniwixErrorCodes.ValidationError)
-            {
-                throw new UniwixValidationException(e.Code, e.Reason, invoiceFile.Content);
-            }
+            var result = await PostAsync<PostInvoiceResponse>(url, content);
+            return result.Map(r => new SendInvoiceResult(r.FileId, r.Message));
         }
 
-        public async Task<InvoiceState> GetInvoiceStateAsync(string fileId)
+        public async Task<ITry<InvoiceState, ErrorResult>> GetInvoiceStateAsync(string fileId)
         {
             var url = $"{UniwixBaseUrl}/Invoices/{fileId}";
             var result = await GetAsync<List<InvoiceStateResult>>(url);
 
-            if (result.Count == 0)
-            {
-                throw new InvalidOperationException($"Invoice {fileId} not found.");
-            }
+            var validatedResult = result.FlatMap(r => r.ToTry(a => a.NonEmpty(), _ => ErrorResult.Create($"Invoice {fileId} not found.", ErrorType.InvoiceNotFound)));
```

Comment:
Isn't there a Where method for this? So you wouldn't have to use the ToTry inside?